Aplib.Core/Intent/Actions/QueryAction.cs

```diff
@@ -25,30 +25,39 @@ public class GuardedAction<TBeliefSet, TQuery> : Action<TBeliefSet>
         /// <summary>
         /// Gets or sets the guard of the action.
         /// </summary>
-        protected new System.Func<TBeliefSet, TQuery?> _guard { get; set; }
+        protected System.Func<TBeliefSet, TQuery?> _query { get; set; }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="GuardedAction{TBeliefSet,TQuery}"/> class.
+        /// Initializes a new instance of the <see cref="QueryAction{TBeliefSet,TQuery}"/> class.
         /// </summary>
         /// <param name="effect">The effect of the action.</param>
-        /// <param name="guard">The guard of the action.</param>
+        /// <param name="query">The guard of the action.</param>
```

Comment:
There are more comments that need to be addressed.
```suggestion
        /// <param name="query">The query of the action.</param>
```